SteamKit2/SteamKit2/Util/HardwareUtils.cs

```diff
@@ -340,17 +400,19 @@ public static void Init()
                 return null;
             }
 
+            if ( !isMachineInfoProviderInitialized )
```

Comment:
This also seems subject to race conditions. Why do we mark it as initialized here, long after the actual initialization?